src/Uri.php

```diff
@@ -31,7 +31,7 @@ public function getScheme(): string
     private ?string $username = null;
     private ?string $password = null;
 
-    public function withUserInfo(string $user, string $password = null): UriInterface
+    public function withUserInfo(?string $user, ?string $password = null): UriInterface
```

Comment:
The parameter $user in the [UriInterface](https://github.com/php-fig/http-message/blob/master/src/UriInterface.php#L208) doesn't accept null

```suggestion
    public function withUserInfo(string $user, ?string $password = null): UriInterface
```

Reply:
Sharp observation, i overlooked that constraint :)